src/ios/PushPlugin.m

```diff
@@ -122,6 +161,27 @@ - (void)init:(CDVInvokedUrlCommand*)command;
      (UIRemoteNotificationTypeBadge | UIRemoteNotificationTypeSound | UIRemoteNotificationTypeAlert)];
 #endif
     
+    //  GCM options
+    [self setGcmSenderId: [NSString stringWithString: [iosOptions objectForKey:@"senderID"]]];
```

Comment:
bah bad checkin - corrected.  At one point I forgot to make the string
defined as strong and was getting corrupted sometimes as you'd expect if
you fail to keep the reference - tried that before I realized real issue
and forgot to undo it.  Did a couple of quick checks with latest build to
make sure it doesnt explode anymore

On Mon, Dec 7, 2015 at 8:06 PM, Simon MacDonald <notifications@github.com>
wrote:

> In src/ios/PushPlugin.m
> <https://github.com/phonegap/phonegap-plugin-push/pull/282#discussion_r46902846>
> :
>
> > @@ -122,6 +161,27 @@ - (void)init:(CDVInvokedUrlCommand*)command;
> >       (UIRemoteNotificationTypeBadge | UIRemoteNotificationTypeSound | UIRemoteNotificationTypeAlert)];
> >  #endif
> >
> > +    //  GCM options
> > +    [self setGcmSenderId: [NSString stringWithString: [iosOptions objectForKey:@"senderID"]]];
>
> @jtbdevelopment <https://github.com/jtbdevelopment> this will cause:
>
> Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[NSPlaceholderString initWithString:]: nil argument'
>
> if the options.ios object doesn't have a senderID object.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/phonegap/phonegap-plugin-push/pull/282/files#r46902846>
> .
>